di/main.py

```diff
@@ -456,3 +458,146 @@ def __get__(self, inst, cls):
             return self.dependencies[self.class_obj]
         except KeyError:
             raise LookupError('Unable to find an instance for {0}'.format(self.class_obj))
+
+
+class ProxyDependencyMap(DependencyMap):
+    """
+    Derivated class that implements the proper __call__ method
+    to return an InjectorProxy .
+    """
+    def __call__(self, key):
+        """ Proxy factory method.
+            >>> dm = ProxyDependencyMap()
+            >>> my_injected_dep = dm(Spam)
+        """
+        return InjectorProxy(key, self)
+
+
+class InjectorProxy(object):
+    """
+    Alternate way of using the injector with a Proxy
+
+        >>> dm = ProxyDependencyMap()
+        >>> myfoo = dm(FOO)
+
+    This code is based on the LocalProxy implemented by the Werkzeug
+    https://github.com/pallets/werkzeug/blob/master/werkzeug/local.py#L254
+    """
+
+    def __init__(self, class_obj, dependencies):
+        object.__setattr__(self, '_InjectorProxy__class_obj', class_obj)
+        object.__setattr__(self, '_InjectorProxy__dependencies', dependencies)
+
+    def _get_current_object(self):
```

Comment:
:nail_care: while it's cool that we use the same method name as in *werkzeug* I wonder if it shouldn't be something much less likely to exist on the proxied value, something like `__get_proxied_dependency` perhaps?